src/utils/generator.js

```diff
@@ -130,16 +134,29 @@ export class Generator {
   emitGlobalDelete(key: string, strictMode: boolean) {
     this._addEntry({
       args: [],
+      bindings: [],
       buildNode: ([]) =>
         t.expressionStatement(t.unaryExpression("delete", this.preludeGenerator.globalReference(key, !strictMode))),
     });
   }
 
+  emitBindingAssignment(binding: Binding, value: Value) {
+    this._addEntry({
+      args: [value],
+      bindings: [binding],
+      buildNode: ([valueNode, bindingNode]) =>
```

Comment:
The `bindings` don't really have to be wired through `_addEntry` / `GeneratorEntry`. `context` is an extra argument passed to `buildNode`, but yet not consumed here. If you add that parameter you, you can just emit the call to `context.serializeBinding(bindingArg)` from within this `buildNode` function, instead of doing it in `_addEntry`. Also, I suggest to rename `serializeBinding` to something like `getReferentializedBindingNode`, as it doesn't actually serialize an actual value.